web/pdf_find_controller.js

```diff
@@ -105,11 +109,51 @@ var PDFFindController = (function PDFFindControllerClosure() {
       });
     },
 
+    // Helper for multiple search.
+    // Sorting array of objects { match: <match>, matchLength: <matchLength> }
+    // in increasing index first and then the lengths.
+    sortMatches: function PDFFindController_sortMatches(matchesWithLength) {
+      matchesWithLength.sort(function(a, b) {
+        return a.match === b.match ?
+          a.matchLength - b.matchLength : a.match - b.match;
+        });
+    },
+
+    // Helper for multiple search - for cases when one search term
+    // include another search term (for example, "tamed tame").
+    // Looking for the same elements in the 'matches' and
+    // leave elements with a longer match-length.
+    // Array matchesWithLength is pre-sorted by method sortMatches.
+    // Example - before:
+    // matches       1 2 2 2 5 7
+    // matchesLength 4 4 5 6 3 2
+    // After:
+    // matches       1 2 5 7
+    // matchesLength 4 6 3 2
+    clearMatches: function PDFFindController_clearMatches(matchesWithLength) {
+      var resultMatches = [], resultMatchesLength = [];
+      var i, len;
+      for (i = 0, len = matchesWithLength.length; i < len; i++) {
+        if (matchesWithLength[i+1] !== undefined &&
+            matchesWithLength[i].match === matchesWithLength[i+1].match) {
+          continue;
+        }
+        resultMatches.push(matchesWithLength[i].match);
+        resultMatchesLength.push(matchesWithLength[i].matchLength);
+      }
+      return { resultMatches: resultMatches,
+               resultMatchesLength: resultMatchesLength };
+    },
+
     calcFindMatch: function PDFFindController_calcFindMatch(pageIndex) {
       var pageContent = this.normalize(this.pageContents[pageIndex]);
       var query = this.normalize(this.state.query);
       var caseSensitive = this.state.caseSensitive;
+      var phraseSearch = this.state.phraseSearch;
       var queryLen = query.length;
+      var matches = [], matchesLength = [];
+      var matchesWithLength = [];
+      var subqueryArray, temp;
```

Comment:
The `temp` variable definition can be moved down to the context where it's used.